ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -164,7 +164,11 @@
                      (api-core/get-user-id-from-token context)
                      subscriber)
         email (if-not email
-                (urs/get-user-email context subscriber)
+                (if-not subscriber
```

Comment:
similar to comment above, you `if` over `if-not`. Don't know why you add this check here. The permission check should have happened before this function is called.

Reply:
This was implemented to handle the specific case where a user tries to ingest a subscription without including a token in the header. In this case, the `subscriber` defined directly before is `nil`, which means we try to get URS user info for user `null` in the urs request.

Reply:
The permission check happens after we call this function, so we would error out by way of URS responding with a 500 before we get to the ingest permission check. Specifically wanted to avoid that case

Reply:
Didn't realize that you are responding to my comments as I was typing it. :)
We could make `get-user-email` more robust by adding handling of a nil subscriber. After some thought, I think throwing exception here is OK if you add a comment.

Reply:
Comment added